packages/expo-google-sign-in/src/index.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+import { NativeModulesProxy } from 'expo-core';
+
+import User from './User';
+
+const { ExpoGoogleSignIn } = NativeModulesProxy;
+
+export const { ERRORS, SCOPES, TYPES } = ExpoGoogleSignIn;
+
+export type GoogleSignInType = TYPES.DEFAULT | TYPES.GAMES;
+
+export type GoogleSignInOptions = {
+  scopes: ?Array<string>,
+  webClientId: ?string,
+  hostedDomain: ?string,
+  accountName: ?string,
+
+  // Android
+  signInType: ?GoogleSignInType,
+  isOfflineEnabled: ?boolean,
+  isPromptEnabled: ?boolean,
+  // iOS
+  clientId: ?string,
+  language: ?string,
+  openIdRealm: ?string,
+};
+
+export type GoogleSignInAuthResult = { type: 'success' | 'cancel', user: ?User };
+
+export class GoogleSignIn {
+  _initialization: Promise;
+  _currentUser: User;
+
+  ERRORS = ERRORS;
+  SCOPES = SCOPES;
+  TYPES = TYPES;
+
+  get currentUser(): User {
+    return this._currentUser;
+  }
+
+  _setCurrentUser(currentUser: ?User): User {
+    this._currentUser = currentUser;
+    return this._currentUser;
+  }
+
+  _validateOptions = (options: ?GoogleSignInOptions): GoogleSignInOptions => {
+    if (!options || !Object.keys(options).length) {
+      throw new Error('GoogleSignIn: you must provide a meaningful configuration, empty provided');
+    }
+    if (options.offlineAccess && !options.webClientId) {
+      throw new Error('GoogleSignIn: Offline access requires server `webClientId`');
+    }
+
+    const DEFAULT_SCOPES = [SCOPES.PROFILE, SCOPES.EMAIL];
+    return {
+      ...options,
+      scopes: options.scopes || DEFAULT_SCOPES,
+    };
+  };
+
+  askForPlayServicesAsync = () => this.hasPlayServicesAsync({ shouldUpdate: true });
+
+  hasPlayServicesAsync = async (options = { shouldUpdate: false }) => {
```

Comment:
what do you think about renaming this to `arePlayServicesAvailableAsync`? I think that might be just a bit more clear that we're talking about the _device_ having play services available